### ISSUE TITLE
Support for JSON snippets

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#591]
(Original issue 884 created by konklone on 2013-07-09T00:01:19.996735+00:00)

(Importing this from [linguist on Github](https://github.com/github/linguist/issues/505))

I'd like to add support for a slight modification of the JSON lexer - that of JSON snippets, where surrounding braces or brackets are allowed to be missing.

In other words, doing this:

```
"first_name": "Jefferson",
"nicknames": ["Jeff", "Bob"],
"last_name": "Brown",
"middle_name": "B.",
"name_suffix": null
```

This is a pattern I use in READMEs and API documentation, when I want to discuss pieces of an object at a time.

Would others find this useful? What's the easiest way to go about doing this? Is there a way to "subclass" the existing JSON lexer and tweak the rules?

**Comment by Anteru** [comment]
(Original issue was assigned to tshatch)

(Original issue 884 last updated on 2016-06-01T05:20:14.474447+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by benatkin on 2014-05-14T06:09:25.706253+00:00)

So much this. [I just added an issue about it to markup.](https://github.com/github/markup/issues/301#issuecomment-43045331) The error characters are really grating.

**Comment by Anteru** [comment]
(Original comment by tshatch on 2016-06-01T05:19:13.799195+00:00)

Add a new lexer that assumes json object is already open.

Fixes #884

→ <<cset a6ffe276bba2>>

**Comment by Anteru** [comment]
(Original comment by tshatch on 2016-06-01T05:19:56.408134+00:00)

I know this doesn't 'help with github anymore, but... fixed.
